tests/test_http_transport.py

```diff
@@ -0,0 +1,290 @@
+"""Tests for HTTP transport functionality."""
+
+import threading
+import time
+from unittest.mock import MagicMock, patch
+
+import pytest
+import requests
+
+from mcp_jupyter.server import create_server
+
+
+class TestHTTPTransport:
+    """Test HTTP transport functionality."""
+
+    def test_create_server_with_default_settings(self):
+        """Test server creation with default settings."""
+        server = create_server()
+        assert server is not None
+        assert server.name == "notebook"
+
+    def test_create_server_with_custom_host_port(self):
+        """Test server creation with custom host and port."""
+        server = create_server(host="0.0.0.0", port=9090)
+        assert server is not None
+        assert server.name == "notebook"
+
+    def test_create_server_with_stateless_mode(self):
+        """Test server creation with stateless HTTP mode."""
+        server = create_server(stateless_http=True)
+        assert server is not None
+        assert server.name == "notebook"
+
+    @pytest.mark.integration
+    def test_http_server_startup(self):
+        """Test that HTTP server can start and respond to requests."""
+        server = create_server(port=8081)
+
+        # Start server in a separate thread
+        server_thread = threading.Thread(
+            target=lambda: server.run(transport="streamable-http"), daemon=True
+        )
+        server_thread.start()
+
+        # Give server time to start
+        time.sleep(2)
+
+        try:
+            # Test that server responds to HTTP requests
+            response = requests.post(
+                "http://127.0.0.1:8081/mcp",
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json, text/event-stream",
+                },
+                json={
+                    "jsonrpc": "2.0",
+                    "method": "initialize",
+                    "params": {
+                        "protocolVersion": "2025-06-18",
+                        "capabilities": {"tools": {}},
+                        "clientInfo": {"name": "test-client", "version": "1.0.0"},
+                    },
+                    "id": 1,
+                },
+                timeout=5,
+            )
+
+            # Check that we get a response (even if it's an error due to missing session)
+            assert response.status_code in [200, 202, 400, 406]
+
+        except requests.exceptions.RequestException:
+            pytest.skip("HTTP server did not start in time")
```

Comment:
Is this the behavior we want? I think probably fine because these are integraiton tests but maybe instead of RequestException, we can throw a timeout exception in our connection loop and skip under that condition instead